### PR TITLE
Added Python wrapping and documentation for alphamat module

### PR DESCRIPTION
**Pull request by sunitanyk**
<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=linux,docs
```

### CODE REVIEW

**Comment by alalek** [comment]
Please rebase patch. Use something like:
```
git fetch upstream
git rebase -i upstream/master
```

**Comment by alalek** [comment]
> Merge pull request from opencv/master

Do not use PRs for updating forks.

This way can't remove trash from patches:

> wants to merge 13 commits

---

Use `git rebase -i upsteam/master` with dropping already "handled" commits instead.

**Comment by sunitanyk** [comment]
Rebased my local branch to upstream/master and then force pushed to origin master.

File: modules/alphamat/include/opencv2/alphamat.hpp [hunk 1]
```diff
@@ -9,19 +9,28 @@
 
 /**
  * @defgroup alphamat Alpha Matting
- * This module is dedicated to compute alpha matting of images, given the input image and an input trimap.
- * The samples directory includes easy examples of how to use the module.
+ * Alpha matting is used to extract a foreground object with soft boundaries from a background image.
+ *
+ * This module is dedicated to computing alpha matte of objects in images from a given input image and a greyscale trimap image that contains information about the foreground, background and unknown pixels. The unknown pixels are assumed to be a combination of foreground and background pixels. The algorithm uses a combination of multiple carefully defined pixels affinities to estimate the opacity of the foreground pixels in the unkown region.
+ *
+ * The implementation is based on "Designing Effective Inter-Pixel Information Flow for Natural Image Matting" by Yağız Aksoy, Tunç Ozan Aydın and Marc Pollefeys, CVPR 2017.
```
Comment (alalek) [inline modules/alphamat/include/opencv2/alphamat.hpp:16]: Please avoid non-ASCII symbols in public headers. For references consider using `@cite` and entry in .bib file.

File: modules/alphamat/tutorials/alphamat_tutorial.markdown [hunk 2]
```diff
@@ -7,15 +7,24 @@ This project was part of Google Summer of Code 2019.
...
-Alphamatting is the problem of extracting the foreground from an image. The extracted foreground can be used for further operations like changing the background in an image.
+Alphamatting is the problem of extracting the foreground with soft boundaries from a background image. The extracted foreground can be used for further operations like changing the background in an image.
 
 Given an input image and its corresponding trimap, we try to extract the foreground from the background. Following is an example:
 
-Input Image: ![](samples/input_images/plant.jpg)
-Input Trimap: ![](samples/trimaps/plant.png)
-Output alpha Matte: ![](samples/output_mattes/plant_result.jpg)
+Input Image: ![](../samples/input_images/plant.jpg)
+Input image should be preferably a RGB image.
```
Comment (alalek) [inline modules/alphamat/tutorials/alphamat_tutorial.markdown:15]: Please check images and formation of the result documentation page [here](http://pullrequest.opencv.org/buildbot/export/pr_contrib/2729/docs/dd/d0e/tutorial_alphamat.html).

See also #2743

Comment (sunitanyk) [reply modules/alphamat/tutorials/alphamat_tutorial.markdown:15]: Updated the image links. The local files image in the same directory as the markdown file are showing fine in GitHub, while the first three ones with path starting from alphamat folder are not. How to check if all image references are working in the final OpenCV doc page?

File: modules/alphamat/include/opencv2/alphamat.hpp (hunk 1 above)
Comment (sunitanyk) [reply modules/alphamat/include/opencv2/alphamat.hpp]: Done

File: modules/alphamat/tutorials/alphamat_tutorial.markdown (hunk 2 above)
Comment (sunitanyk) [reply modules/alphamat/tutorials/alphamat_tutorial.markdown:15]: I can also move the samples folder images inside tutorial, if separate samples folder is not necessary. Please let me know.

Comment (sunitanyk) [reply modules/alphamat/tutorials/alphamat_tutorial.markdown:15]: Just refreshed in the [test url ](https://pullrequest.opencv.org/buildbot/export/pr_contrib/2729/docs/dd/d0e/tutorial_alphamat.html) you gave. All images look good now. 👍

Comment (alalek) [reply modules/alphamat/tutorials/alphamat_tutorial.markdown:15]: It is OK to keep images in `samples`, which can be used as sample's input.

**Review by alalek — APPROVED** [review]
Looks great! Thank you :+1:

Possible next steps are:
- adding Python sample
- adding test for Python (as `misc/python/test/test_alphamat.py`)
- also there is a way to improve tutorial by adding multiple languages support ([example](https://docs.opencv.org/4.5.0/d3/d96/tutorial_basic_geometric_drawing.html))

**Comment by sunitanyk** [comment]
@alalek I can do the Python example and test file changes you suggested above in this PR, if you leave it open. Or I can create a new one, if you need to close this one.

**Comment by alalek** [comment]
This PR is merged.
Please update your fork (rewrite master branch), create new branch (avoid using "master" branch name for patches), prepare changes and open a new PR.

**Pull request merged**